lib/updatePrograms.js

```diff
@@ -16,70 +19,73 @@
  * and HTML just the code
  */
 
-const firestore = require("../firebase");
-
 const updatePrograms = async function(uid, json) {
+  winston.info(`Starting async user programs update`);
+  winston.info(`UID: ${uid}`);
+  winston.verbose(`JSON: ${JSON.stringify(json || {})}`);
+
   if (!json) {
+    winston.error("No JSON provided, Exiting");
     return { ok: false, error: "No json provided" };
   }
   if (!uid) {
+    winston.error("No UID provided, Exiting");
     return { ok: false, error: "No uid provided" };
   }
 
   try {
-    console.log("Updating programs for uid:" + uid);
     //create document with base user data at /users/:uid
     const collection = firestore
       .collection("/users")
       .doc(uid.toString())
       .collection("programs");
 
     const programs = Object.keys(json);
+
     if (!programs.length) {
+      winston.error(`Provided JSON has no keys (user ${uid}). Exiting`);
       return { ok: false, error: "json provided has no keys" };
     }
+
     let promises = [];
 
+    //function to be called on every program in the QuerySnapshot programs
+    const updateProgram = async program => {
+      let doc = {};
+      try {
+        doc = await firestore
+          .collection("/users")
+          .doc(uid.toString())
+          .collection("programs")
+          .doc(program)
+          .get();
+      } catch (err) {
+        winston.error(
+          `"${program}" program for ${uid} not found, failed to get doc. Received error: ${err}...`,
+        );
+      }
+
+      if (doc.exists) {
+        winston.verbose(`Updating '${program}'...`);
+        winston.verbose(`Setting value to '${JSON.stringify(json[program] || {})}'`);
+        return collection.doc(program).update(json[program]);
```

Comment:
Here, this is erroring on user save - however, looking at the [Firebase Docs on this matter](https://firebase.google.com/docs/reference/js/firebase.firestore.DocumentReference#update), I believe that it should be fine to run update with one parameter. 

Here's the error for reference:

```
error: Failed to update programs for SMInip4YSaWgn0LEjBuAkRywuHL2. Received error: FirebaseError: [code=invalid-argument]: Function DocumentReference.update() requires at least 2 arguments, but was called with 1 argument. [2019-03-27T00:57:23.133Z]
```

Reply:
Decided to solve this in the frontend repo rather than alter the functionality of the BE